themes/default/content/blog/project-config-mvp/index.md

```diff
@@ -0,0 +1,97 @@
+---
+title: "Simpler configuration management with project level config"
+date: 2022-11-02
+meta_desc: The introduction of project level config makes infrastructure configuration even easier with Pulumi.
+meta_image: meta.png
+authors:
+    - fraser-waters
+    - zaid-ajaj
+tags:
+    - features
+    - config
+---
+
+One of our most up-voted feature requests (with 78 thumbs ups) is to [support hierarchical config](https://github.com/pulumi/pulumi/issues/2307). We're happy to announce that we've now released the first part of plans to support this feature.
+
+Pulumi will now allow you to set configuration values in your `Pulumi.yaml` file, using the given value as a default for all stacks in the project. While we expect even this first level of support will be incredibly useful to many people we also want to assure you that we have many more plans in place to make this feature even better.
+
+### Project level configuration values
+
+Pulumi has always had the concept of [configuration per stack](/docs/intro/concepts/config/), and the changes for project level configuration haven't fundamentally changed that. You can still set config for stacks via the the `pulumi config` CLI command, or by editing the `Pulumi.<stack>.yaml` file. Project level config just gives you the possibility to set default values for all stacks in your `Pulumi.yaml`.
+
+A common request we get for this is making sure that all stacks use the same AWS region, or don't have default providers enabled. You can now do this by setting these in your `Pulumi.yaml`.
+
+```yaml
+name: project
+runtime: go
+config:
+    pulumi:disable-default-providers: [“*”]
+    aws:region: us-east-1
+```
+
+As you can see this allows setting default configuration values per-stack for configuring the engine or providers, but you can also set default values for project configuration. For example maybe you have a default count of the number of virtual machines to create:
+
+```yaml
+name: project
+runtime: go
+config:
+    vm-count: 2
+```
+
+The only caveat to mention about this system, and it's related to the next section where we're going to discuss specifications, is that if you have object values they must be nested under a key `value`. So if you want to set the key `obj` to the object value `{ foo: "bar" }` you would need to write it as such:
+
+```yaml
+name: project
+runtime: go
+config:
+    obj:
+        value:
+            foo: bar
+```
+
+This is to allow the engine to distinguish between values and schemas, which will show below.
+
+### Configuration specification
+
+A related feature to defining values for config at the project level is [being able to define a schema for the config](https://github.com/pulumi/pulumi/issues/1052). This allows you to define what configuration settings you expect in your `Pulumi.yaml`, the engine can then use that information to validate configuration for every stack.
```

Comment:
```suggestion
A related feature to defining values for config at the project level is [being able to define a schema for the config](https://github.com/pulumi/pulumi/issues/1052). This allows you to define what configuration settings you expect in your `Pulumi.yaml` so that the engine can use that information to validate configuration for every stack.
```